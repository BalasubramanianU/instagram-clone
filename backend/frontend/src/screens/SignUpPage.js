import React, { useState, useEffect } from "react";
import "../css/styles.css";
import { Link } from "react-router-dom";
import {
  validateEmail,
  validateName,
  validateNumber,
  validatePassword,
} from "../utils/userValidation";

const SignUpPage = () => {
  const [formData, setFormData] = useState({
    logInId: "",
    fullName: "",
    userName: "",
    password: "",
  });
  const [isValid, setIsValid] = useState(false);
  const [passwordField, setPasswordField] = useState({
    inputType: "password",
    button: "Show",
  });
  const [error, setError] = useState({});

  useEffect(() => {
    if (
      formData.logInId.length > 0 &&
      formData.fullName.length > 0 &&
      formData.userName.length > 0 &&
      formData.password.length > 5
    ) {
      return setIsValid(true);
    }
    setIsValid(false);
  }, [formData]);

  const handleChange = (event) => {
    setFormData({ ...formData, [event.target.name]: event.target.value });
  };

  const passwordButtonClick = () => {
    passwordField.button === "Show"
      ? setPasswordField({ inputType: "text", button: "Hide" })
      : setPasswordField({ inputType: "password", button: "Show" });
  };

  const handleSubmit = () => {
    const apiCall = async (loginType) => {
      if (!loginType) return;
      const requestOptions = {
        method: "POST",
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify({
          [loginType]: formData.logInId,
          fullName: formData.fullName,
          userName: formData.userName,
          password: formData.password,
        }),
      };
      try {
        setIsValid(false);
        const response = await fetch(
          // "http://192.168.1.8:5000/user/signup",
          "https://insta-clone-bala.herokuapp.com/user/signup",
          requestOptions
        );
        if (response.ok) {
          localStorage.token = response.headers.get("x-auth-header");
          setIsValid(true);
          return;
        }
        const errorMessage = await response.text();
        if (errorMessage.includes("user already exists")) {
          if (Number.isInteger(Number(formData.logInId)))
            setError({ mobileNumber: true });
          else setError({ email: true });
        }
        if (errorMessage.includes("User name already exists"))
          setError({ userName: true });
        setIsValid(true);
      } catch (error) {
        // all the errors in the try block will accumulate here, you need to
        // write logic here if you want to handle connection failed and other
        // exceptions/errors seperately.
        console.log(error);
        setIsValid(true);
      }
    };
    if (validatePassword(formData.password).error) return;
    if (validateName(formData.userName).error) return;
    if (!validateEmail(formData.logInId).error) apiCall("email");
    else if (!validateNumber(formData.logInId).error) apiCall("mobileNumber");
    else {
      if (Number.isInteger(Number(formData.logInId)))
        return setError({ mobileNumber: true });
      else return setError({ email: true });
    }
    setError({});
  };

  return (
    <div className="backgroundLayer">
      <div className="headerSignUp"></div>
      <div className="gridContainerSignUp">
        <div className="signUpBox">
          <div className="logo"></div>
          <p className="headerText">
            Sign up to see photos and videos from your friends.
          </p>
          <button className="signUpButton">
            <div className="iconContainerSignUp">
              <div className="fbIconSignUp">
                <ion-icon name="logo-facebook"></ion-icon>
              </div>
            </div>
            Log in with Facebook
          </button>
          <div className="orComponent">
            <div className="lineStrike"></div>
            <p className="orText">OR</p>
            <div className="lineStrike"></div>
          </div>
          <div className="inputField">
            <input
              type="text"
              minLength="1"
              maxLength="255"
              placeholder="Mobile Number or Email"
              required
              name="logInId"
              value={formData.logInId}
              onChange={handleChange}
            />
            <span className="nameSpan">Mobile Number or Email</span>
          </div>
          <div className="inputField">
            <input
              type="text"
              minLength="1"
              maxLength="255"
              placeholder="Full Name"
              required
              name="fullName"
              value={formData.fullName}
              onChange={handleChange}
            />
            <span className="nameSpan">Full Name</span>
          </div>
          <div className="inputField">
            <input
              type="text"
              minLength="1"
              maxLength="255"
              placeholder="Username"
              required
              name="userName"
              value={formData.userName}
              onChange={handleChange}
            />
            <span className="nameSpan">Username</span>
          </div>
          <div className="inputField">
            <input
              type={passwordField.inputType}
              minLength="1"
              maxLength="1024"
              placeholder="Password"
              required
              name="password"
              value={formData.password}
              onChange={handleChange}
            />
            <span className="passwordSpan">Password</span>
            <button onClick={passwordButtonClick} testId="passwordButton">
              {passwordField.button}
            </button>
          </div>
          <div className="buttonContainer">
            <button
              className="button"
              onClick={handleSubmit}
              disabled={!isValid}
              testId="signupButton"
            >
              Sign up
            </button>
          </div>
          {error.email && (
            <span className="errorMessage">Enter a valid email address.</span>
          )}
          {error.mobileNumber && (
            <span className="errorMessage">
              Looks like your phone number may be incorrect. Please try entering
              your full number, including the country code.
            </span>
          )}
          {error.userName && (
            <span className="errorMessage">
              This username isn't available. Please try another.
            </span>
          )}

          <p className="smallText">
            By signing up, you agree to our <b>Terms , Data Policy </b>and
            <b>Cookies Policy .</b>
          </p>
        </div>
        <div className="switchAuthMethodBoxSignUp">
          <p className="generalText">
            Have an account?
            <Link className="login" to="/login">
              Log in
            </Link>
          </p>
        </div>
        <div className="appStoreContainerSignUp">
          <p className="generalText">Get the app.</p>
          <div className="appStoreImages">
            <button className="googlePlay"></button>
            <button className="appStore"></button>
          </div>
        </div>
      </div>
      <div className="footerContainer">
        <div className="footer">
          <div className="footerFirstRow">
            <a className="footerText" href>
              About
            </a>
            <a className="footerText" href>
              Blog
            </a>
            <a className="footerText" href>
              Jobs
            </a>
            <a className="footerText" href>
              Help
            </a>
            <a className="footerText" href>
              API
            </a>
            <a className="footerText" href>
              Privacy
            </a>
            <a className="footerText" href>
              Terms
            </a>
            <a className="footerText" href>
              Top Accounts
            </a>
            <a className="footerText" href>
              Hashtags
            </a>
            <a className="footerText" href>
              Locations
            </a>
            <a className="footerText" href>
              Instagram Lite
            </a>
          </div>
          <div className="footerSecondRow">
            <select className="footerDropdown">
              <option>English</option>
            </select>
            <p className="smallText">&copy 2021 Instagram from Facebook</p>
          </div>
        </div>
      </div>
    </div>
  );
};

export default SignUpPage;
