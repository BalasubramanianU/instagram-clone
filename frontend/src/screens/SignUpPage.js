import React, { useState, useEffect } from "react";
import "../css/styles.css";
import { Link } from "react-router-dom";

const SignUpPage = () => {
  const [formData, setFormData] = useState({
    logInId: "",
    fullName: "",
    userName: "",
    password: "",
  });
  const [isValid, setIsValid] = useState(false);
  const [passwordField, setPasswordField] = useState({
    inputType: "password",
    button: "Show",
  });

  useEffect(() => {
    if (
      formData.logInId.length > 0 &&
      formData.fullName.length > 0 &&
      formData.userName.length > 0 &&
      formData.password.length > 5
    ) {
      return setIsValid(true);
    }
    setIsValid(false);
  }, [formData]);

  const handleChange = (event) => {
    setFormData({ ...formData, [event.target.name]: event.target.value });
  };

  const passwordButtonClick = () => {
    passwordField.button === "Show"
      ? setPasswordField({ inputType: "text", button: "Hide" })
      : setPasswordField({ inputType: "password", button: "Show" });
  };

  const handleSubmit = () => {
    console.log(formData);
    // TODO:to be continued...
    // const requestOptions = {
    //   method: "POST",
    //   headers: { "Content-Type": "application/json" },
    //   body: JSON.stringify({
    //     email: formData.logInId,
    //     fullName: formData.fullName,
    //     userName: formData.userName,
    //     password: formData.password,
    //   }),
    // };
    // fetch("http://locahost:5000/user/signup", requestOptions).then(
    //   (res) => console.log(res.json()),
    //   (error) => console.log(error)
    // );
  };

  return (
    <div className="backgroundLayer">
      <div className="headerSignUp"></div>
      <div className="gridContainerSignUp">
        <div className="signUpBox">
          <div className="logo"></div>
          <p className="headerText">
            Sign up to see photos and videos from your friends.
          </p>
          <button className="signUpButton">
            <div className="iconContainerSignUp">
              <div className="fbIconSignUp">
                <ion-icon name="logo-facebook"></ion-icon>
              </div>
            </div>
            Log in with Facebook
          </button>
          <div className="orComponent">
            <div className="lineStrike"></div>
            <p className="orText">OR</p>
            <div className="lineStrike"></div>
          </div>
          <div className="inputField">
            <input
              type="text"
              minLength="1"
              maxLength="255"
              placeholder="Mobile Number or Email"
              required
              name="logInId"
              value={formData.logInId}
              onChange={handleChange}
            />
            <span className="nameSpan">Mobile Number or Email</span>
          </div>
          <div className="inputField">
            <input
              type="text"
              minLength="1"
              maxLength="255"
              placeholder="Full Name"
              required
              name="fullName"
              value={formData.fullName}
              onChange={handleChange}
            />
            <span className="nameSpan">Full Name</span>
          </div>
          <div className="inputField">
            <input
              type="text"
              minLength="1"
              maxLength="255"
              placeholder="Username"
              required
              name="userName"
              value={formData.userName}
              onChange={handleChange}
            />
            <span className="nameSpan">Username</span>
          </div>
          <div className="inputField">
            <input
              type="password"
              minLength="1"
              maxLength="1024"
              placeholder="Password"
              required
              name="password"
              value={formData.password}
              onChange={handleChange}
            />
            <span className="passwordSpan">Password</span>
            <button onClick={passwordButtonClick}>
              {passwordField.button}
            </button>
          </div>
          <div className="buttonContainer">
            <button
              className="button"
              onClick={handleSubmit}
              disabled={!isValid}
            >
              Sign up
            </button>
          </div>
          <span className="errorMessageHidden">
            Sorry, your password was incorrect. Please double-check your
            password.
          </span>
          <p className="smallText">
            By signing up, you agree to our <b>Terms , Data Policy </b>and
            <b>Cookies Policy .</b>
          </p>
        </div>
        <div className="switchAuthMethodBoxSignUp">
          <p className="generalText">
            Have an account?
            <Link className="login" to="/login">
              Log in
            </Link>
          </p>
        </div>
        <div className="appStoreContainerSignUp">
          <p className="generalText">Get the app.</p>
          <div className="appStoreImages">
            <button className="googlePlay"></button>
            <button className="appStore"></button>
          </div>
        </div>
      </div>
      <div className="footerContainer">
        <div className="footer">
          <div className="footerFirstRow">
            <a className="footerText" href>
              About
            </a>
            <a className="footerText" href>
              Blog
            </a>
            <a className="footerText" href>
              Jobs
            </a>
            <a className="footerText" href>
              Help
            </a>
            <a className="footerText" href>
              API
            </a>
            <a className="footerText" href>
              Privacy
            </a>
            <a className="footerText" href>
              Terms
            </a>
            <a className="footerText" href>
              Top Accounts
            </a>
            <a className="footerText" href>
              Hashtags
            </a>
            <a className="footerText" href>
              Locations
            </a>
            <a className="footerText" href>
              Instagram Lite
            </a>
          </div>
          <div className="footerSecondRow">
            <select className="footerDropdown">
              <option>English</option>
            </select>
            <p className="smallText">&copy 2021 Instagram from Facebook</p>
          </div>
        </div>
      </div>
    </div>
  );
};

export default SignUpPage;
