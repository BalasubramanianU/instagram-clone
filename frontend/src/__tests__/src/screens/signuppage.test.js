import { BrowserRouter } from "react-router-dom";
import { create, act } from "react-test-renderer";
import SignUpPage from "../../../screens/SignUpPage";

describe("input and button scenarios", () => {
  let tree, instance, element;
  beforeAll(() => {
    tree = create(
      <BrowserRouter>
        <SignUpPage />
      </BrowserRouter>
    );
    instance = tree.root;
  });

  it("should render the singup page correctly", () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("should change the input field type if show button is clicked on password field", () => {
    element = instance.findByProps({ testId: "passwordButton" });
    act(() => {
      element.props.onClick();
    });
    element = instance.findByProps({ name: "password" });

    expect(element.props.type).toBe("text");
  });

  it("should change the input field type if hide button is clicked on password field", () => {
    element = instance.findByProps({ testId: "passwordButton" });
    act(() => {
      element.props.onClick();
    });
    element = instance.findByProps({ name: "password" });

    expect(element.props.type).toBe("password");
  });

  it("should disable singup button if input values are empty in input field", () => {
    element = instance.findByProps({ testId: "signupButton" });

    expect(element.props.disabled).toBe(true);
  });

  it("should not disable singup button if valid values are present in the input field", () => {
    act(() => {
      element = instance.findByProps({ name: "logInId" });
      element.props.onChange({ target: { name: "logInId", value: "1234" } });
    });
    act(() => {
      element = instance.findByProps({ name: "fullName" });
      element.props.onChange({ target: { name: "fullName", value: "bala" } });
    });
    act(() => {
      element = instance.findByProps({ name: "userName" });
      element.props.onChange({ target: { name: "userName", value: "bala_" } });
    });
    act(() => {
      element = instance.findByProps({ name: "password" });
      element.props.onChange({ target: { name: "password", value: "abcdef" } });
    });
    element = instance.findByProps({ testId: "signupButton" });

    expect(element.props.disabled).toBe(false);
  });

  it("should display a error message when user logs in using invalid user id - email", () => {
    act(() => {
      element = instance.findByProps({ name: "logInId" });
      element.props.onChange({ target: { name: "logInId", value: "ab" } });
    });
    act(() => {
      element = instance.findByProps({ name: "fullName" });
      element.props.onChange({ target: { name: "fullName", value: "bala" } });
    });
    act(() => {
      element = instance.findByProps({ name: "userName" });
      element.props.onChange({ target: { name: "userName", value: "bala_" } });
    });
    act(() => {
      element = instance.findByProps({ name: "password" });
      element.props.onChange({ target: { name: "password", value: "abcdef" } });
    });
    act(() => {
      element = instance.findByProps({ testId: "signupButton" });
      element.props.onClick();
    });
    element = instance.findAllByType("span");
    const errorMsgLength = element.filter(
      (spanElement) =>
        spanElement.props.children === "Enter a valid email address."
    ).length;

    expect(errorMsgLength).toBe(1);
  });

  it("should display a error message when user logs in using invalid user id - mobile number", () => {
    act(() => {
      element = instance.findByProps({ name: "logInId" });
      element.props.onChange({ target: { name: "logInId", value: "12" } });
    });
    act(() => {
      element = instance.findByProps({ name: "fullName" });
      element.props.onChange({ target: { name: "fullName", value: "bala" } });
    });
    act(() => {
      element = instance.findByProps({ name: "userName" });
      element.props.onChange({ target: { name: "userName", value: "bala_" } });
    });
    act(() => {
      element = instance.findByProps({ name: "password" });
      element.props.onChange({ target: { name: "password", value: "abcdef" } });
    });
    act(() => {
      element = instance.findByProps({ testId: "signupButton" });
      element.props.onClick();
    });
    element = instance.findAllByType("span");
    const errorMsgLength = element.filter(
      (spanElement) =>
        spanElement.props.children ===
        "Looks like your phone number may be incorrect. Please try entering your full number, including the country code."
    ).length;

    expect(errorMsgLength).toBe(1);
  });

  it("should not display a error message when user logs in using invalid password", () => {
    act(() => {
      element = instance.findByProps({ name: "logInId" });
      element.props.onChange({ target: { name: "logInId", value: "12" } });
    });
    act(() => {
      element = instance.findByProps({ name: "fullName" });
      element.props.onChange({ target: { name: "fullName", value: "bala" } });
    });
    act(() => {
      element = instance.findByProps({ name: "userName" });
      element.props.onChange({ target: { name: "userName", value: "bala_" } });
    });
    act(() => {
      element = instance.findByProps({ name: "password" });
      element.props.onChange({ target: { name: "password", value: "abcd" } });
    });
    act(() => {
      element = instance.findByProps({ testId: "signupButton" });
      element.props.onClick();
    });
    element = instance.findAllByType("span");
    const errorMsgLength = element.filter(
      (spanElement) =>
        spanElement.props.children ===
        "Sorry, your password was incorrect. Please double-check your password."
    ).length;

    expect(errorMsgLength).toBe(0);
  });

  it("should not display a error message when user logs in using invalid user name", () => {
    act(() => {
      element = instance.findByProps({ name: "logInId" });
      element.props.onChange({ target: { name: "logInId", value: "12" } });
    });
    act(() => {
      element = instance.findByProps({ name: "fullName" });
      element.props.onChange({ target: { name: "fullName", value: "bala" } });
    });
    act(() => {
      element = instance.findByProps({ name: "userName" });
      element.props.onChange({ target: { name: "userName", value: 1 } });
    });
    act(() => {
      element = instance.findByProps({ name: "password" });
      element.props.onChange({ target: { name: "password", value: "abcdef" } });
    });
    act(() => {
      element = instance.findByProps({ testId: "signupButton" });
      element.props.onClick();
    });
    element = instance.findAllByType("span");
    const errorMsgLength = element.filter(
      (spanElement) =>
        spanElement.props.children ===
        "This username isn't available. Please try another."
    ).length;

    expect(errorMsgLength).toBe(0);
  });
});

describe("api call - positive scenarios", () => {
  let tree,
    instance,
    element,
    token = "json web token here";
  beforeAll(() => {
    tree = create(
      <BrowserRouter>
        <SignUpPage />
      </BrowserRouter>
    );
    instance = tree.root;
  });

  afterEach(() => {
    localStorage.clear();
  });

  it("should receive a jwt token if singup request is successful with mobileNumber", async () => {
    jest.spyOn(global, "fetch").mockResolvedValue({
      ok: true,
      headers: {
        get: jest.fn().mockReturnValue(token),
      },
      json: jest.fn().mockResolvedValue({ mobileNumber: "1234" }),
    });
    act(() => {
      element = instance.findByProps({ name: "logInId" });
      element.props.onChange({ target: { name: "logInId", value: "1234" } });
    });
    act(() => {
      element = instance.findByProps({ name: "userName" });
      element.props.onChange({ target: { name: "userName", value: "bala_" } });
    });
    act(() => {
      element = instance.findByProps({ name: "password" });
      element.props.onChange({ target: { name: "password", value: "abcdef" } });
    });
    act(() => {
      element = instance.findByProps({ name: "password" });
      element.props.onChange({ target: { name: "password", value: "abcdef" } });
    });
    await act(async () => {
      element = instance.findByProps({ testId: "signupButton" });
      element.props.onClick();
    });

    expect(localStorage.getItem("token")).toBe(token);
  });

  it("should receive a jwt token if singup request is successful with email", async () => {
    jest.spyOn(global, "fetch").mockResolvedValue({
      ok: true,
      headers: {
        get: jest.fn().mockReturnValue(token),
      },
      json: jest.fn().mockResolvedValue({ email: "abc@abc.com" }),
    });
    act(() => {
      element = instance.findByProps({ name: "logInId" });
      element.props.onChange({
        target: { name: "logInId", value: "abc@abc.com" },
      });
    });
    act(() => {
      element = instance.findByProps({ name: "userName" });
      element.props.onChange({ target: { name: "userName", value: "bala_" } });
    });
    act(() => {
      element = instance.findByProps({ name: "password" });
      element.props.onChange({ target: { name: "password", value: "abcdef" } });
    });
    act(() => {
      element = instance.findByProps({ name: "password" });
      element.props.onChange({ target: { name: "password", value: "abcdef" } });
    });
    await act(async () => {
      element = instance.findByProps({ testId: "signupButton" });
      element.props.onClick();
    });

    expect(localStorage.getItem("token")).toBe(token);
  });
});

describe("api call - negative scenarios", () => {
  let tree, instance, element;
  beforeAll(() => {
    tree = create(
      <BrowserRouter>
        <SignUpPage />
      </BrowserRouter>
    );
    instance = tree.root;
  });

  let mockResponse = (response) => {
    response
      ? jest.spyOn(global, "fetch").mockResolvedValue({
          text: jest.fn().mockResolvedValue(response),
        })
      : jest.spyOn(global, "fetch").mockRejectedValue("connection error");
  };

  it("should display a error message when user logs in using invalid user id - mobile number", async () => {
    mockResponse("user already exists");
    act(() => {
      element = instance.findByProps({ name: "logInId" });
      element.props.onChange({ target: { name: "logInId", value: "1234" } });
    });
    act(() => {
      element = instance.findByProps({ name: "userName" });
      element.props.onChange({ target: { name: "userName", value: "bala_" } });
    });
    act(() => {
      element = instance.findByProps({ name: "password" });
      element.props.onChange({ target: { name: "password", value: "abcdef" } });
    });
    act(() => {
      element = instance.findByProps({ name: "password" });
      element.props.onChange({ target: { name: "password", value: "abcdef" } });
    });
    await act(async () => {
      element = instance.findByProps({ testId: "signupButton" });
      element.props.onClick();
    });
    element = instance.findAllByType("span");
    const errorMsgLength = element.filter(
      (spanElement) =>
        spanElement.props.children ===
        "Looks like your phone number may be incorrect. Please try entering your full number, including the country code."
    ).length;

    expect(errorMsgLength).toBe(1);
    expect(localStorage.getItem("token")).toBe(null);
  });

  it("should display a error message when user logs in using invalid user id - email", async () => {
    mockResponse("user already exists");
    act(() => {
      element = instance.findByProps({ name: "logInId" });
      element.props.onChange({
        target: { name: "logInId", value: "abc@gmail.com" },
      });
    });
    act(() => {
      element = instance.findByProps({ name: "userName" });
      element.props.onChange({ target: { name: "userName", value: "bala_" } });
    });
    act(() => {
      element = instance.findByProps({ name: "password" });
      element.props.onChange({ target: { name: "password", value: "abcdef" } });
    });
    act(() => {
      element = instance.findByProps({ name: "password" });
      element.props.onChange({ target: { name: "password", value: "abcdef" } });
    });
    await act(async () => {
      element = instance.findByProps({ testId: "signupButton" });
      element.props.onClick();
    });
    element = instance.findAllByType("span");
    const errorMsgLength = element.filter(
      (spanElement) =>
        spanElement.props.children === "Enter a valid email address."
    ).length;

    expect(errorMsgLength).toBe(1);
    expect(localStorage.getItem("token")).toBe(null);
  });

  it("should display a error message when user logs in using invalid user name", async () => {
    mockResponse("User name already exists");
    act(() => {
      element = instance.findByProps({ name: "logInId" });
      element.props.onChange({ target: { name: "logInId", value: "1234" } });
    });
    act(() => {
      element = instance.findByProps({ name: "userName" });
      element.props.onChange({ target: { name: "userName", value: "bala_" } });
    });
    act(() => {
      element = instance.findByProps({ name: "password" });
      element.props.onChange({ target: { name: "password", value: "abcdef" } });
    });
    act(() => {
      element = instance.findByProps({ name: "password" });
      element.props.onChange({ target: { name: "password", value: "abcdef" } });
    });
    await act(async () => {
      element = instance.findByProps({ testId: "signupButton" });
      element.props.onClick();
    });
    element = instance.findAllByType("span");
    const errorMsgLength = element.filter(
      (spanElement) =>
        spanElement.props.children ===
        "This username isn't available. Please try another."
    ).length;

    expect(errorMsgLength).toBe(1);
    expect(localStorage.getItem("token")).toBe(null);
  });

  it("should display the error message in console when api request fails", async () => {
    mockResponse();
    const consoleSpy = jest.spyOn(console, "log");
    act(() => {
      element = instance.findByProps({ name: "logInId" });
      element.props.onChange({ target: { name: "logInId", value: "1234" } });
    });
    act(() => {
      element = instance.findByProps({ name: "userName" });
      element.props.onChange({ target: { name: "userName", value: "bala_" } });
    });
    act(() => {
      element = instance.findByProps({ name: "password" });
      element.props.onChange({ target: { name: "password", value: "abcdef" } });
    });
    act(() => {
      element = instance.findByProps({ name: "password" });
      element.props.onChange({ target: { name: "password", value: "abcdef" } });
    });
    await act(async () => {
      element = instance.findByProps({ testId: "signupButton" });
      element.props.onClick();
    });

    expect(consoleSpy).toHaveBeenCalledWith("connection error");
  });
});
