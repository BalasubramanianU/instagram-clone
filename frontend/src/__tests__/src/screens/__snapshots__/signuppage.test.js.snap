// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input and button scenarios should render the login page correctly 1`] = `
<div
  className="backgroundLayer"
>
  <div
    className="headerSignUp"
  />
  <div
    className="gridContainerSignUp"
  >
    <div
      className="signUpBox"
    >
      <div
        className="logo"
      />
      <p
        className="headerText"
      >
        Sign up to see photos and videos from your friends.
      </p>
      <button
        className="signUpButton"
      >
        <div
          className="iconContainerSignUp"
        >
          <div
            className="fbIconSignUp"
          >
            <ion-icon
              name="logo-facebook"
            />
          </div>
        </div>
        Log in with Facebook
      </button>
      <div
        className="orComponent"
      >
        <div
          className="lineStrike"
        />
        <p
          className="orText"
        >
          OR
        </p>
        <div
          className="lineStrike"
        />
      </div>
      <div
        className="inputField"
      >
        <input
          maxLength="255"
          minLength="1"
          name="logInId"
          onChange={[Function]}
          placeholder="Mobile Number or Email"
          required={true}
          type="text"
          value=""
        />
        <span
          className="nameSpan"
        >
          Mobile Number or Email
        </span>
      </div>
      <div
        className="inputField"
      >
        <input
          maxLength="255"
          minLength="1"
          name="fullName"
          onChange={[Function]}
          placeholder="Full Name"
          required={true}
          type="text"
          value=""
        />
        <span
          className="nameSpan"
        >
          Full Name
        </span>
      </div>
      <div
        className="inputField"
      >
        <input
          maxLength="255"
          minLength="1"
          name="userName"
          onChange={[Function]}
          placeholder="Username"
          required={true}
          type="text"
          value=""
        />
        <span
          className="nameSpan"
        >
          Username
        </span>
      </div>
      <div
        className="inputField"
      >
        <input
          maxLength="1024"
          minLength="1"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
        <span
          className="passwordSpan"
        >
          Password
        </span>
        <button
          onClick={[Function]}
          testId="passwordButton"
        >
          Show
        </button>
      </div>
      <div
        className="buttonContainer"
      >
        <button
          className="button"
          disabled={true}
          onClick={[Function]}
          testId="signupButton"
        >
          Sign up
        </button>
      </div>
      <p
        className="smallText"
      >
        By signing up, you agree to our 
        <b>
          Terms , Data Policy 
        </b>
        and
        <b>
          Cookies Policy .
        </b>
      </p>
    </div>
    <div
      className="switchAuthMethodBoxSignUp"
    >
      <p
        className="generalText"
      >
        Have an account?
        <a
          className="login"
          href="/login"
          onClick={[Function]}
        >
          Log in
        </a>
      </p>
    </div>
    <div
      className="appStoreContainerSignUp"
    >
      <p
        className="generalText"
      >
        Get the app.
      </p>
      <div
        className="appStoreImages"
      >
        <button
          className="googlePlay"
        />
        <button
          className="appStore"
        />
      </div>
    </div>
  </div>
  <div
    className="footerContainer"
  >
    <div
      className="footer"
    >
      <div
        className="footerFirstRow"
      >
        <a
          className="footerText"
          href={true}
        >
          About
        </a>
        <a
          className="footerText"
          href={true}
        >
          Blog
        </a>
        <a
          className="footerText"
          href={true}
        >
          Jobs
        </a>
        <a
          className="footerText"
          href={true}
        >
          Help
        </a>
        <a
          className="footerText"
          href={true}
        >
          API
        </a>
        <a
          className="footerText"
          href={true}
        >
          Privacy
        </a>
        <a
          className="footerText"
          href={true}
        >
          Terms
        </a>
        <a
          className="footerText"
          href={true}
        >
          Top Accounts
        </a>
        <a
          className="footerText"
          href={true}
        >
          Hashtags
        </a>
        <a
          className="footerText"
          href={true}
        >
          Locations
        </a>
        <a
          className="footerText"
          href={true}
        >
          Instagram Lite
        </a>
      </div>
      <div
        className="footerSecondRow"
      >
        <select
          className="footerDropdown"
        >
          <option>
            English
          </option>
        </select>
        <p
          className="smallText"
        >
          &copy 2021 Instagram from Facebook
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`input and button scenarios should render the singup page correctly 1`] = `
<div
  className="backgroundLayer"
>
  <div
    className="headerSignUp"
  />
  <div
    className="gridContainerSignUp"
  >
    <div
      className="signUpBox"
    >
      <div
        className="logo"
      />
      <p
        className="headerText"
      >
        Sign up to see photos and videos from your friends.
      </p>
      <button
        className="signUpButton"
      >
        <div
          className="iconContainerSignUp"
        >
          <div
            className="fbIconSignUp"
          >
            <ion-icon
              name="logo-facebook"
            />
          </div>
        </div>
        Log in with Facebook
      </button>
      <div
        className="orComponent"
      >
        <div
          className="lineStrike"
        />
        <p
          className="orText"
        >
          OR
        </p>
        <div
          className="lineStrike"
        />
      </div>
      <div
        className="inputField"
      >
        <input
          maxLength="255"
          minLength="1"
          name="logInId"
          onChange={[Function]}
          placeholder="Mobile Number or Email"
          required={true}
          type="text"
          value=""
        />
        <span
          className="nameSpan"
        >
          Mobile Number or Email
        </span>
      </div>
      <div
        className="inputField"
      >
        <input
          maxLength="255"
          minLength="1"
          name="fullName"
          onChange={[Function]}
          placeholder="Full Name"
          required={true}
          type="text"
          value=""
        />
        <span
          className="nameSpan"
        >
          Full Name
        </span>
      </div>
      <div
        className="inputField"
      >
        <input
          maxLength="255"
          minLength="1"
          name="userName"
          onChange={[Function]}
          placeholder="Username"
          required={true}
          type="text"
          value=""
        />
        <span
          className="nameSpan"
        >
          Username
        </span>
      </div>
      <div
        className="inputField"
      >
        <input
          maxLength="1024"
          minLength="1"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
        <span
          className="passwordSpan"
        >
          Password
        </span>
        <button
          onClick={[Function]}
          testId="passwordButton"
        >
          Show
        </button>
      </div>
      <div
        className="buttonContainer"
      >
        <button
          className="button"
          disabled={true}
          onClick={[Function]}
          testId="signupButton"
        >
          Sign up
        </button>
      </div>
      <p
        className="smallText"
      >
        By signing up, you agree to our 
        <b>
          Terms , Data Policy 
        </b>
        and
        <b>
          Cookies Policy .
        </b>
      </p>
    </div>
    <div
      className="switchAuthMethodBoxSignUp"
    >
      <p
        className="generalText"
      >
        Have an account?
        <a
          className="login"
          href="/login"
          onClick={[Function]}
        >
          Log in
        </a>
      </p>
    </div>
    <div
      className="appStoreContainerSignUp"
    >
      <p
        className="generalText"
      >
        Get the app.
      </p>
      <div
        className="appStoreImages"
      >
        <button
          className="googlePlay"
        />
        <button
          className="appStore"
        />
      </div>
    </div>
  </div>
  <div
    className="footerContainer"
  >
    <div
      className="footer"
    >
      <div
        className="footerFirstRow"
      >
        <a
          className="footerText"
          href={true}
        >
          About
        </a>
        <a
          className="footerText"
          href={true}
        >
          Blog
        </a>
        <a
          className="footerText"
          href={true}
        >
          Jobs
        </a>
        <a
          className="footerText"
          href={true}
        >
          Help
        </a>
        <a
          className="footerText"
          href={true}
        >
          API
        </a>
        <a
          className="footerText"
          href={true}
        >
          Privacy
        </a>
        <a
          className="footerText"
          href={true}
        >
          Terms
        </a>
        <a
          className="footerText"
          href={true}
        >
          Top Accounts
        </a>
        <a
          className="footerText"
          href={true}
        >
          Hashtags
        </a>
        <a
          className="footerText"
          href={true}
        >
          Locations
        </a>
        <a
          className="footerText"
          href={true}
        >
          Instagram Lite
        </a>
      </div>
      <div
        className="footerSecondRow"
      >
        <select
          className="footerDropdown"
        >
          <option>
            English
          </option>
        </select>
        <p
          className="smallText"
        >
          &copy 2021 Instagram from Facebook
        </p>
      </div>
    </div>
  </div>
</div>
`;
